src/Nox.Types/Types/Yaml/Yaml.cs

```diff
@@ -0,0 +1,35 @@
+using System.IO;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+
+namespace Nox.Types;
+
+/// <summary>
+///  Represents a Nox <see cref="Yaml"/> type and value object.
+/// </summary>
+public class Yaml : ValueObject<string, Yaml>
+{
+    /// <summary>
+    /// Validates the YAML object and returns the validation result.
+    /// </summary>
+    /// <returns>The validation result.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        try
+        {
+            var deserializer = new DeserializerBuilder().Build();
```

Comment:
Can we have the deserializer as a static field, so that it's not created each time we create a Yaml type?

Reply:
Requried changes were applied